pages/api/tokenData/static.ts

```diff
@@ -0,0 +1,109 @@
+
+type ResponseData = {
+  hash: string
+  binary_value: string
+  binary_attributes: BinaryAttribute[]
+  type: string
+  phrase_value: string
+  phrase_attributes: BinaryAttribute[]
+};
+
+function getHashBinaryAttributes(hash: string): BinaryAttribute[] {
+  function getAttrValue(attr: Attribute, hash: string) {
+    if (attr.regex) {
+      const hex2binVal = hex2bin(hash).match(attr.regex);
+      const binaryAttrLength = hex2binVal !== null ? hex2binVal[0].length : 0;
+      return binaryAttrLength;
+    }
+
+    if (attr.calculationFunction) return attr.calculationFunction(hash);
+
+    return null;
+  }
+
+  return HASH_ATTRIBUTES.map((attribute) => ({
+    trait_type: attribute.descriptionShort,
+    value: getAttrValue(attribute, hash)
+  }));
+}
+
+function getPhraseRarity(str: string, arr: ethers.Event[]): number {
+  const allPhrases = arr.map(event => event?.args?.phrase);
+  const phraseAmount = allPhrases.filter((phrase: string) => phrase === str).length;
+  return phraseAmount / allPhrases.length * 100;
+}
+
+function getPhraseAttributes(
+  phrase: string,
+  events: ethers.Event[]
+): BinaryAttribute[] {
+  const phraseWords: string[] = phrase.split(' ');
+  const phraseCharCount: number = phraseWords.reduce((prev: number, curr: string) => prev + curr.length, 0);
+  return [
+    {
+      trait_type: 'rarity',
+      value: getPhraseRarity(phrase, events)
+    },
+    {
+      trait_type: 'word amount',
+      value: phraseWords.length
+    },
+    {
+      trait_type: 'character amount',
+      value: phraseCharCount
+    }
+  ];
+}
+
+export default async function handler(
+  req: NextApiRequest,
+  res: NextApiResponse<ResponseData | string>
+) {
+  const { tokenId } = req.query;
+
+  if (isNaN(Number(tokenId))) {
+    res.status(400).send('tokenId must be a number');
+    return;
+  }
+
+  const hashesContract = getHashesContract(1);
+
+  const [hash, isDeactivated]: [string, boolean] = await Promise.all([
+    hashesContract.getHash(tokenId),
+    hashesContract.deactivated(tokenId),
+  ]);
+
+  if (hash === ethers.constants.HashZero) {
+    res.status(404).send('token not found');
+    return;
+  }
+
+  const binaryValue = hex2bin(hash);
+  const binaryAttributes = getHashBinaryAttributes(hash);
+
+  //TODO: temp phrase solution for now
+  const generatedFilter = hashesContract.filters.Generated();
+  const AllGeneratedEvents = await hashesContract.queryFilter(generatedFilter);
+  const tokenIdEvent = AllGeneratedEvents.find(event => Number(event?.args?.tokenId) === Number(tokenId));
+
+  const phrase = tokenIdEvent?.args?.phrase;
+  const phraseAttributes = phrase ? getPhraseAttributes(phrase, AllGeneratedEvents) : [];
+
+  res.status(200).json({
+    hash,
+    binary_value: binaryValue,
+    binary_attributes: binaryAttributes,
+    type: Number(tokenId) >= 1000 ? 'Standard' : isDeactivated ? 'DAO Deactivated' : 'DAO',
+    phrase_value: phrase || 'Phrase is being non-fungibilitized... check back soon!',
```

Comment:
you wouldn't need this check if you're using the events. It only really matters if we're pulling from a DB which has a job where it gets written to every 15 mins (this covers the case where the token exists but the job hasnt run yet)